### DISCUSSION TITLE
Step-by-step guide to migrate custom auth provider in custom verifier from web3auth's side

### DISCUSSION THREAD

**Discussion by qqqllppp** [Web3Auth/Web3Auth#436] [Help]
Currently I am using firebase as my custom auth provider in my custom verifier. However before committing, I want to know if its possible to move away from firebase to another custom auth provider (non-google) and plug that in the same custom verifier (basically if I would be stuck in firebase?). This would mean that I export the auth data of my users from firebase into the new database with its own auth system, and plug this in to the existing web3auth custom verifier. 

The expected behaviour after migration is:
1. My users are able to login the dapp as usual using the email they used to sign-up even before the migration happened
2. using this email, the same old web3 address is connected (not a new address created)

So the export of data and new data base + custom auth would be done on my side, but to safely plug in to an existing web3auth verifier, how would this be done?

**Comment by YZhenY** [comment]
Its possible but nuanced. Web3Auth complies with your app/firebase via the JWT OAuth standard, which is dependent on two main things wrt to your application:

1. Your JWK/keys that you register with Web3Auth

They either have to be the same, or you'll need to update your verifier to the new set of keys that you have setup.

2. the sub field/claim in your JWT

This MUST be the same from Firebase relative to your new auth setup.

If condition 1. and 2. are fufilled you'll be able to migrate your auth provider.
